core/class/jMQTTCmd.class.php

```diff
@@ -246,6 +246,36 @@ public function preSave() {
 
         }
 
+		// Reset auto_publish if info cmd
+		if ($this->getType() == 'info' && $this->getConfiguration('auto_publish', 0))
+			$this->setConfiguration('auto_publish', 0);
+		// Check "request" value if auto_publish enabled
+		if ($this->getType() == 'action' && $this->getConfiguration('auto_publish', 0)) {
+			$req = $this->getConfiguration('request', '');
+			// Get all commands
+			preg_match_all("/#([0-9]*)#/", $req, $matches);
+			$cmds = array_unique($matches[1]);
+			// $value = '';
+			if (count($cmds) > 0) { // There are commands
+				foreach ($cmds as $cmd_id) {
+					$cmd = cmd::byId($cmd_id);
+					if (!is_object($cmd))
+						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd_id.'</b> est invalide.');
+					if ($cmd->getType() != 'info')
+						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> n\'est pas de type info.');
+					if ($cmd->getEqType() =='jMQTT' && $this->getTopic() == $cmd->getTopic())
+						throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> référence le même topic.');
+					// if ($cmd->getEqType() =='jMQTT' && $this->getEqLogic()->getBrkId() == $cmd->getEqLogic()->getBrkId())
+						// throw new Exception('Impossible d\'activer la publication automatique sur <b>'.$this->getHumanName().'</b> car la commande <b>'.$cmd->getHumanName().'</b> appartient au même Broker.');
+					// $value .= '#'.$cmd->getId().'#';
+				}
+				// $this->setValue($value);
+			} else {// Reset auto_publish if no command
+				$this->setConfiguration('auto_publish', 0);
```

Comment:
End User will go crazy :-D

Reply:
Should I throw an Exception in that case ?

Reply:
I would say : do nothing, do not create a listener

Reply:
Ok, handled at jMQTTCmd::preRemove()